theano/tensor/signal/pool.py

```diff
@@ -189,9 +187,9 @@ def out_shape(imgshape, ds, ignore_border=False, st=None, padding=(0, 0)):
         r = tensor.extract_constant(r)
         c = tensor.extract_constant(c)
         if padding[0]:
-            r += padding[0] * 2
+            r = r + padding[0] * 2
```

Comment:
If I remember correctly it was because it was raising an error otherwise. The `+=` was not working in one of the cases, because the type of variables was somehow not compatible.